ansible_ai_connect/ai/api/utils/segment.py

```diff
@@ -118,11 +120,32 @@ def base_send_segment_event(
                     "event_name": event_name,
                     "msg_len": msg_len,
                 },
-                "timestamp": event["timestamp"],
+                "timestamp": event.timestamp,
             }
             send_segment_event(event, "segmentError", user)
 
 
+def init_schema1_client() -> None:
+    def on_segment_error(error, _):
+        logger.error(f"An error occurred in sending schema1 data to Segment: {error}")
+
+    if settings.SEGMENT_WRITE_KEY:
+        if not analytics.write_key:
+            analytics.write_key = settings.SEGMENT_WRITE_KEY
+            analytics.debug = settings.DEBUG
+            analytics.gzip = True  # Enable gzip compression
+            # analytics.send = False # for code development only
+            analytics.on_error = on_segment_error
+
+
+def send_schema1_event(event_obj) -> None:
+    print(f"SENDING SCHEMA1 EVENT (name={event_obj.event_name})\n{event_obj.as_dict()} ({type(event_obj)})")
```

Comment:
Use a logger vs `print`?

Reply:
This is a debug trace will be removed later.